server/src/test/java/io/crate/integrationtests/TransportSQLActionTest.java

```diff
@@ -1938,6 +1938,26 @@ public void test_bit_string_can_be_inserted_and_queried() throws Exception {
         );
     }
 
+    @Test
+    @UseRandomizedSchema(random = false)
+    public void test_bit_string_operators_in_where_clause() throws Exception {
+        execute("CREATE TABLE employees (name text, status BIT(3), comment TEXT)");
+        execute("INSERT INTO employees VALUES ('Anna', B'100','Anna works remote')");
+        execute("refresh table employees");
+
+        execute("SELECT * FROM employees WHERE status & B'100' = B'100'");
+        assertThat(response).hasRows(
+            "Anna| B'100'| Anna works remote\n"
+        );
+    }
+
+    @Test
+    @UseRandomizedSchema(random = false)
+    public void test_bit_string_with_different_length() throws Exception {
+        execute("CREATE TABLE tbl (b BIT(3))");
+        execute("SELECT * FROM tbl WHERE b = B'1'");
+    }
+
```

Comment:
Is having `SignatureBinderTest` enough? If so, can drop those tests

Reply:
I think a `SignatureBinderTest` and maybe an `ExpressionAnalyzerTest` similar to https://github.com/crate/crate/blob/master/server/src/test/java/io/crate/analyze/expressions/ExpressionAnalyzerTest.java#L380 is enough an much preferred over an integration test. (Also something like `SELECT B'110' = B'1'; ` would be enough to test the issue, no need for a real table).

Reply:
added `ExpressionAnalyzerTest`  and removed Itest in https://github.com/crate/crate/pull/13425/commits/35278524ebe821dbcc36511677953bb5d66f9994